twisted/conch/ssh/keys.py

```diff
@@ -782,18 +783,23 @@ def public(self):
         return Key(self._keyObject.public_key())
 
 
-    def fingerprint(self):
+    def fingerprint(self, hash_function=sha256):
         """
         Get the user presentation of the fingerprint of this L{Key}.  As
         described by U{RFC 4716 section
         4<http://tools.ietf.org/html/rfc4716#section-4>}::
 
-            The fingerprint of a public key consists of the output of the MD5
-            message-digest algorithm [RFC1321].  The input to the algorithm is
-            the public key data as specified by [RFC4253].  (...)  The output
-            of the (MD5) algorithm is presented to the user as a sequence of 16
-            octets printed as hexadecimal with lowercase letters and separated
-            by colons.
+            The fingerprint of a public key consists of the output of the
+            message-digest algorithm for the given hash function(sha256 or MD5).
+            The input to the algorithm is the public key data as specified by [RFC4253].
+            The output of sha256 (default)[RFC4634] algorithm is presented to the
+            user in the form of base64 encoded sha256 hashes.
+            The output of the MD5[RFC1321] algorithm is presented to the user as
+            a sequence of 16 octets printed as hexadecimal with lowercase letters
+            and separated by colons.
```

Comment:
You can't change this text like this. The paragraph above notes that this is a quote from RFC 4716, which you have now edited.

You can either remove the quote entirely, or you can add new text below this quote, but you can't change it.